### PR TITLE
docs: add an integration strategy guide

### PR DESCRIPTION
**Pull request by gmontard**
## Description
Adding a guide on the "integration strategy" of Bearer in developer's worklflow.

### CODE REVIEW

File: docs/guides/integration-strategy.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+If your code triggers any findings, as show above, read the next section. Otherwise, skip directly to step 3.
+
+## Step 2: Triage and remediate existing findings
+
+We need to start triaging findings from the CLI on your local machine. 
+
+Here is our recommended strategy:
+1. Start by addressing "Critical" findings, then "High", "Medium", and finally "Low" findings.
+2. Remove any irrelevant findings for each severity level by using the ```bearer ignore``` command and categorize them as *false positive* (see [doc](/guides/configure-scan/#ignore-specific-findings)).
+3. If a rule is problematic with your code, skip it entirely using the ```--skip-rule``` option (see [doc](/guides/configure-scan/#skip-rules-for-the-entire-scan)).
```
Comment (elsapet) [inline docs/guides/integration-strategy.md]: Small suggestion - reads a bit clearer for me but you can take it or leave it. 
```suggestion
3. If a rule is problematic for your codebase, skip it entirely using the ```--skip-rule``` option (see [doc](/guides/configure-scan/#skip-rules-for-the-entire-scan)).
```

File: docs/guides/integration-strategy.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Step 3: Minimize new issues from being introduced
+
+To prevent the introduction of new issues in your codebase, it is crucial to identify and address them before developers merge their code into the main branch, as part of your CI.
+
+Start configuring a Bearer scan when developers create a new PR/MR. This allows for immediate feedback and provides the necessary context for any detected findings:
+- For GitHub Actions, refer to [using GitHub Action](/guides/github-action/#pull-request-diff).
+- For GitLab CI, refer to the [using GitLab Action](/guides/gitlab/#gitlab-merge-request-diff).
+- For CircleCI, refer to [using CircleCI](/guides/ci-setup/#circleci).
+- For other CI, refer to the [universel setup](/guides/ci-setup/#universal-setup).
```
Comment (elsapet) [inline docs/guides/integration-strategy.md]:
```suggestion
- For other CI, refer to the [universal setup](/guides/ci-setup/#universal-setup).
```

File: docs/guides/integration-strategy.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+However, some findings may persist. This could be because they haven't been addressed during the previous step (*if you don't enforce PR/PR checks*), or because a complete code scan is required to identify specific findings spanning across unmodified code.
+
+To prevent significant security vulnerabilities in production, we recommend conducting a full scan as part of your CD process or when merging code into the main branch. Since breaking deployments is a serious matter, we advise limiting scanning to certain severity levels, such as critical, or to specific rules using the ```--only-rule``` option (see [doc](/guides/configure-scan/#run-only-specified-rules)).
+
+
+
+## Step 5: Schedule regular complete scans
+
+If you've followed our previous setup recommendations, you may have noticed that certain findings will persist in production. This is particularly true for low-priority findings or rules that you have low confidence in and disabled.
```
Comment (elsapet) [inline docs/guides/integration-strategy.md]:
```suggestion
If you've followed our previous setup recommendations, you may have noticed that certain findings will persist in production. This is particularly true for low-priority findings or rules that you have low confidence in and have disabled.
```

File: docs/guides/integration-strategy.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+To address these findings and maintain effective control over your security posture, we recommend conducting a full scan of your codebase at least once a month. This scan should allow all severity levels and rules to ensure a comprehensive review.
```
Comment (elsapet) [inline docs/guides/integration-strategy.md]: A suggestion to emphasise re-enabling rules - for you to take or leave 👍 

```suggestion
To address these findings and maintain effective control over your security posture, we recommend conducting a full scan of your codebase at least once a month. This scan should allow all severity levels and enable all rules to ensure a comprehensive review.
```

**Review by elsapet — APPROVED** [review]
Small typos and some take-or-leave suggestions :)

**Comment by gmontard** [comment]
> Small typos and some take-or-leave suggestions :)

thank you!!!

**Review by gotbadger — APPROVED** [review]
looking good

**Pull request merged**